src/Compilers/Core/Portable/SourceGeneration/SyntaxStore.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal class SyntaxStore
```

Comment:
>SyntaxStore [](http://example.com/codeflow?start=19&length=11)

How is this type used? (It looks like it has no methods other than a constructor and no public fields.)

Reply:
Its opaque, but the builder accesses its internal members to determine the state. Basically you pass it back to the builder when creating it.

I've flipped it around to make the builder `ctor` private and give `SyntaxStore` a `ToBuilder()` method, which will hopefully make it a little clearer.